core/src/blockchain/headerchain.rs

```diff
@@ -190,8 +190,10 @@ impl HeaderChain {
         match best_header_changed {
             BestHeaderChanged::None => (),
             BestHeaderChanged::CanonChainAppended {
-                ..
+                best_header,
             } => {
+                let header_view = HeaderView::new(best_header);
+                hashes.insert(header_view.number(), header_view.hash());
                 hashes.insert(new_header.number(), new_header.hash());
```

Comment:
Isn't `new_header` the highest header?
Is it okay that `new_hash_entries` returns a hash of the highest one?

Reply:
Removing the `new_header` is included in this commit(https://github.com/CodeChain-io/codechain/pull/1096/commits/43de1349ad32bf8e0a5e5fe8164f5b28183a156b).

Reply:
How about merging it with 43de134?